Geometry_Engine/Compute/CurveProximity.cs

```diff
@@ -0,0 +1,1004 @@
+                    {
+                        result.Item1 = fitPoints[0];
+                        result.Item2 = curve2.ClosestPoint(fitPoints[0]);
+                        check = true;
+                    }
+                    else
+                    {
+                        result.Item1 = fitPoints[1];
+                        result.Item2 = curve2.ClosestPoint(fitPoints[1]);
+                        check = true;
+                    }
+                }
+               
+            }
+            if ((fitPoints = curve2.PlaneIntersections(ftPln1)).Count > 0)
+            {
+                if (check)
+                {
+                    if (fitPoints.Count == 1 && (fitPoints[0].Distance(curve1) < result.Item1.Distance(curve2)))
+                    {
+                        result.Item2 = fitPoints[0];
+                        result.Item1 = curve1.ClosestPoint(fitPoints[0]);
+                    }
+                    else if (fitPoints.Count > 1)
+                    {
+                        if (fitPoints[0].Distance(curve1) < fitPoints[1].Distance(curve1))
+                        {
+                            if (fitPoints[0].Distance(curve1) < result.Item1.Distance(curve2))
+                            {
+                                result.Item2 = fitPoints[0];
+                                result.Item1 = curve1.ClosestPoint(fitPoints[0]);
+                            }
+                        }
+                        else
+                        {
+                            if (fitPoints[1].Distance(curve1) < result.Item1.Distance(curve2))
+                            {
+                                result.Item2 = fitPoints[1];
+                                result.Item1 = curve1.ClosestPoint(fitPoints[1]);
+                            }
+                        }
+                    }
+                }
+                else
+                if (fitPoints.Count == 1)
+                {
+                    result.Item2 = fitPoints[0];
+                    result.Item1 = curve1.ClosestPoint(fitPoints[0]);
+                    check = true;
+                }
+                else if (fitPoints.Count > 1)
+                {
+                    if (fitPoints[0].Distance(curve1) < fitPoints[1].Distance(curve1))
+                    {
+                        
+                            result.Item2 = fitPoints[0];
+                            result.Item1 = curve1.ClosestPoint(fitPoints[0]);
+                            check = true;
+                    }
+                    else
+                    {
+                            result.Item2 = fitPoints[1];
+                            result.Item1 = curve1.ClosestPoint(fitPoints[1]);
+                            check = true;
+                    }
+                }
+            }
+            if (check)
+            {
+                do
+                {
+                    oldresult.Item1 = result.Item1.Clone();
```

Comment:
Are all `Clone` calls necessary?

Reply:
I actually think that yes, I want to pass an object, not exactly a reference to an object. If I passed there result.item1, and then change result.item1, this would also modify oldresult.item1, as oldresult was only having a reference to an object, not a "physical" (funny word in programming) object.

Reply:
But I will check if they are necessary, and if not, then will delete them.

Reply:
Update 1.
They aren't.